wp-parsely.php

```diff
@@ -22,20 +22,29 @@
  * Requires WP:       4.0.0
  */
 
-require 'src/class-parsely.php';
+// If this file is called directly, abort.
+if ( ! defined( 'WPINC' ) ) {
+	die;
+}
 
 if ( class_exists( 'Parsely' ) ) {
-	define( 'PARSELY_VERSION', Parsely::VERSION );
-	if ( ! defined( 'PARSELY_PLUGIN_DIR' ) ) {
-		define( 'PARSELY_PLUGIN_DIR', plugin_dir_path( __FILE__ ) );
-	}
-	if ( ! defined( 'PARSELY_PLUGIN_URL' ) ) {
-		define( 'PARSELY_PLUGIN_URL', plugin_dir_url( __FILE__ ) );
-	}
-	$parsely = new Parsely();
+	return;
+}
+
+if ( ! defined( 'PARSELY_PLUGIN_DIR' ) ) {
+	define( 'PARSELY_PLUGIN_DIR', plugin_dir_path( __FILE__ ) );
+}
+if ( ! defined( 'PARSELY_PLUGIN_URL' ) ) {
+	define( 'PARSELY_PLUGIN_URL', plugin_dir_url( __FILE__ ) );
 }
 
-require 'src/class-parsely-recommended-widget.php';
+require PARSELY_PLUGIN_DIR . 'src/class-parsely.php';
+
+define( 'PARSELY_VERSION', Parsely::VERSION );
+
+$GLOBALS['parsely'] = new Parsely();
```

Comment:
This class may be instantiated when the file is read, as opposed to at `init` or `plugins_loaded` hook which would then allows other plugins to register some action or filter callbacks before the instantiation.

The issue we have right now is that the Parsely constructor does too much (#280).